### PR TITLE
feat(shell): add a util to validate the input ip string with list_nodes

### PR DESCRIPTION
**Pull request by Smityz**
This util can help us if the input IP string is sent to the right replicas belongs to metas.

### CODE REVIEW

File: src/shell/command_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include "command_utils.h"
```
Comment (acelyc111) [inline src/shell/command_utils.cpp:1]: License

File: src/shell/command_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            not_find_ip = false;
+        }
+    }
+
+    if (not_find_ip) {
+        err_info = fmt::format("invalid ip:port:{}, can't find it in metas \n", ip_str);
+        return false;
+    }
+
+    return true;
```
Comment (acelyc111) [inline src/shell/command_utils.cpp:32]:
```suggestion
    if (!target_address.from_string_ipv4(ip_str.c_str())) {
        err_info = fmt::format("invalid ip:port={}, can't transform it into rpc_address", ip_str);
        return false;
    }

    std::map<dsn::rpc_address, dsn::replication::node_status::type> nodes;
    auto error = sc->ddl_client->list_nodes(::dsn::replication::node_status::NS_INVALID, nodes);
    if (error != dsn::ERR_OK) {
        err_info = fmt::format("list nodes failed, error={}", error.to_string());
        return false;
    }

    for (const auto &node : nodes) {
        if (target_address == node.first) {
            return true;
        }
    }

    err_info = fmt::format("invalid ip:port={}, can't find it in metas", ip_str);
    return false;
```

**Review by acelyc111 — DISMISSED** [review]

File: src/shell/command_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                 /*out*/ dsn::rpc_address &target_address,
+                 /*out*/ std::string &err_info)
+{
+    if (!target_address.from_string_ipv4(ip_str.c_str())) {
+        err_info = fmt::format("invalid ip:port={}, can't transform it into rpc_address", ip_str);
+        return false;
+    }
+
+    std::map<dsn::rpc_address, dsn::replication::node_status::type> nodes;
+    auto error = sc->ddl_client->list_nodes(::dsn::replication::node_status::NS_INVALID, nodes);
```
Comment (levy5307) [inline src/shell/command_utils.cpp:31]: remove the prefix `::`

File: src/shell/command_utils.h [hunk 4]
```diff
@@ -9,6 +9,7 @@
 #include <set>
 
 #include "shell/argh.h"
+#include "command_executor.h"
```
Comment (levy5307) [inline src/shell/command_utils.h:12]: remove `#include "command_executor.h"`
using forward declaration is ok

File: src/shell/commands/detect_hotkey.cpp [hunk 5]
```diff
@@ -85,13 +85,13 @@ bool detect_hotkey(command_executor *e, shell_context *sc, arguments args)
     }
 
     dsn::rpc_address target_address;
+    std::string err_info;
     std::string ip_str = cmd({"-d", "--address"}).str();
-    if (!target_address.from_string_ipv4(ip_str.c_str())) {
-        fmt::print("invalid ip, error={}\n", ip_str);
+    if (!validate_ip(sc, ip_str, target_address, err_info)) {
+        fmt::print("{}\n", err_info);
```
Comment (levy5307) [inline src/shell/commands/detect_hotkey.cpp:91]: `fmt::print(stderr, "{}\n", err_info)`

File: src/shell/command_utils.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        return false;
+    }
+
+    for (const auto &node : nodes) {
+        if (target_address == node.first) {
+            return true;
+        }
+    }
+
+    err_info = fmt::format("invalid ip:port={}, can't find it in metas", ip_str);
```
Comment (levy5307) [inline src/shell/command_utils.cpp:45]: can't find it in cluster

**Review by levy5307 — APPROVED** [review]

**Review by acelyc111 — APPROVED** [review]

**Pull request merged**